test/unit_tests/unittest_monitoring_frame_serialization_deserialization.cpp

```diff
@@ -0,0 +1,198 @@
+#include "psen_scan_v2/istring_stream_builder.h"
+
+using namespace psen_scan_v2;
+
+namespace psen_scan_v2_test
+{
+TEST(FieldHeaderTest, testGetIdAndLength)
+{
+  uint8_t id = 5;
+  uint16_t length = 7;
+  FieldHeader header(id, length);
+  EXPECT_EQ(id, header.id());
+  EXPECT_EQ(length, header.length());
+}
+
+TEST(FieldHeaderTest, testReadSuccess)
+{
+  uint8_t id = 5;
+  uint16_t length = 7;
+  uint16_t expected_length = length - 1;
+  uint16_t max_num_bytes = 9;
+
+  IStringStreamBuilder builder;
+  builder.add(id);
+  builder.add(length);
+  std::istringstream is{ builder.get() };
+
+  std::unique_ptr<FieldHeader> header_ptr;
+  ASSERT_NO_THROW(header_ptr.reset(new FieldHeader{ readFieldHeader(is, max_num_bytes) }););
+  EXPECT_EQ(id, header_ptr->id());
+  EXPECT_EQ(expected_length, header_ptr->length());
+}
+
+TEST(FieldHeaderTest, testReadHeaderTooShortFailure)
+{
+  uint16_t too_short_header;
+  uint16_t max_num_bytes = 2;
+
+  IStringStreamBuilder builder;
+  builder.add(too_short_header);
+  std::istringstream is{ builder.get() };
+
+  EXPECT_THROW(readFieldHeader(is, max_num_bytes);, raw_processing::StringStreamFailure);
+}
+
+TEST(MonitoringFrameSerializationTest, testUDPFrameTestDataWithoutIntensitiesSuccess)
+{
+  // Load testdata from dump
+  UDPFrameTestDataWithoutIntensities test_data;
+  DynamicSizeRawData serialized_monitoring_frame_message = serialize(test_data.msg_);
+
+  EXPECT_EQ(test_data.hex_dump.size(), serialized_monitoring_frame_message.size());
+
+  for (size_t i = 0; i < test_data.hex_dump.size(); i++)
+  {
+    EXPECT_EQ((uint8_t)serialized_monitoring_frame_message.at(i), test_data.hex_dump.at(i)) << " index " << i;
+  }
+}
+
+TEST(MonitoringFrameSerializationTest, testUDPFrameTestDataWithoutMeasurementsAndIntensitiesSuccess)
+{
+  // Load testdata from dump
+  UDPFrameTestDataWithoutMeasurementsAndIntensities test_data;
+  DynamicSizeRawData serialized_monitoring_frame_message = serialize(test_data.msg_);
+
+  EXPECT_EQ(test_data.hex_dump.size(), serialized_monitoring_frame_message.size());
+
+  for (size_t i = 0; i < test_data.hex_dump.size(); i++)
+  {
+    EXPECT_EQ((uint8_t)serialized_monitoring_frame_message.at(i), test_data.hex_dump.at(i)) << " index " << i;
+  }
+}
+
+TEST(MonitoringFrameSerializationTest, testSerializationInvariance)
+{
+  UDPFrameTestDataWithoutIntensities test_data;
+  DynamicSizeRawData raw = serialize(test_data.msg_);
+
+  MonitoringFrameMsg deserialized_msg = deserialize(convertToMaxSizeRawData(raw), raw.size());
+
+  EXPECT_EQ(test_data.msg_, deserialized_msg);
+}
+
+TEST(MonitoringFrameSerializationTest, testSerializationInvariance2)
+{
+  MonitoringFrameMsg msg(TenthOfDegree(25), TenthOfDegree(1), 456, { 10, 20, 30, 40 });
+
+  DynamicSizeRawData raw = serialize(msg);
+
+  MonitoringFrameMsg deserialized_msg = deserialize(convertToMaxSizeRawData(raw), raw.size());
+
+  EXPECT_EQ(msg, deserialized_msg);
+}
+
+class MonitoringFrameMsgDeserializeTest : public ::testing::Test
```

Comment:
I would move the deserialize-tests into a new file.

Reply:
Actually I suggested it this way. Mainly because on part of the testing is the invariance test ( msg == deserialize(serialize(msg)) )